sklearn/neighbors/_quad_tree.pyx

```diff
@@ -575,7 +575,7 @@ cdef class _QuadTree:
         if capacity == self.capacity and self.cells != NULL:
             return 0
 
-        if capacity == SIZE_MAX:
+        if <size_t> capacity == SIZE_MAX:
```

Comment:
Another alternative which is:

```python
        if capacity == <SIZE_t> SIZE_MAX:
```

but I think what you chose brings a correct conversion: `capacity` is `SIZE_t` AKA `cnp.npy_intp` AKA `Py_intptr_t` AKA ["an integer that is the size of a pointer on the platform"](https://github.com/numpy/numpy/blob/c1697e0df90a69ab53892e47e33aa809c1b92b29/doc/source/reference/c-api/dtype.rst#L366) whilst, [`SIZE_MAX` is the maximum value for `uint64`](https://github.com/embeddedartistry/libc/blob/40bcc1e14f2e7ffcdda4fe20d6a83b61ade0b86f/include/stdint.h#L249) stored as a `uint64` I suppose.

I find using `SIZE_MAX` for this encoding weird. Would there be configuration for which this would break? Previously, was `capacity` cast to a `size_t` (likely a `uint64` on most architecture nowadays) or was `SIZE_MAX` cast to a `Py_intptr_t`?

@OmarManzoor: can you provide the warning you get on your platform?

Reply:
This is the precise warning:
` sklearn/neighbors/_quad_tree.c:8389:34: warning: comparison of integers of different signs: '__pyx_t_7sklearn_9neighbors_10_quad_tree_SIZE_t' (aka 'long') and 'unsigned long' [-Wsign-compare]
      __pyx_t_1 = ((__pyx_v_capacity == SIZE_MAX) != 0);`